src/main/java/core/basesyntax/model/io/DataConverterImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.model.io;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (String line : lines) {
+            String[] parts = line.split(",");
+
+            if (parts.length != 3) {
+                continue;
```

Malformed input lines are silently skipped here (`if (parts.length != 3) { continue; }`). The specification requires trimming and validation of fields and rejecting malformed lines with clear exceptions: "Trim and validate fields; reject malformed lines with clear exceptions." Instead of continuing, throw a RuntimeException with the offending line and a message explaining the format error. Silent skipping hides input problems.